tests/foreman/api/test_location.py

```diff
@@ -35,10 +26,12 @@ def test_create_location_with_different_names(self, name):
         @Feature: Location
 
         """
-        location = entities.Location(name=name).create()
-        self.assertEqual(location.name, name)
+        for name in valid_data_list():
+            with self.subTest(name):
+                location = entities.Location(name=name).create()
+                self.assertEqual(location.name, name)
 
-    def test_create_location_with_description(self):
+    def test_create_loc_with_desc(self):
```

Comment:
I like the function names or test names to be short and crisp.  Anything more can be added in the docstring as necessary.

And IMHO all the shortforms used here - loc, env, cr, desc - are all familiar ones and it should not affect readability :)